modules/order/commerce_order.module

```diff
@@ -221,6 +221,13 @@ function commerce_order_entity_operation_alter(array &$operations, EntityInterfa
   if ($entity->getEntityTypeId() == 'field_config' && $entity->id() == 'profile.customer.address') {
     unset($operations['storage-settings']);
   }
+
+  // Remove the default delete button to dis-allow the user from deleting
+  // profiles as we want to preserve the order references. The delete will be
+  // replaced by the deactivate action.
+  if ($entity->bundle() === 'customer') {
+    unset($operations['delete']);
+  }
```

Comment:
I'm thinking this should be ideally done in `ProfileAccessControlHandler` where access to the `delete` operation is checked. If the operations or the forms are built properly, the delete operation/button would not be made available in the first place; better than building them first and removing them later. Also, it would disable deleting from everywhere that it might be added in the future making it less bug prone.

Problem is that this is specific to `commerce` and the handler is located on the `profile` module. I don't think it's wise to override the access handler. I think it's worth including it in the pull request to `profile` asking for integration with `commerce`, I think it's acceptable.